source/dev_guide/project-standards.rst

```diff
@@ -0,0 +1,154 @@
+4. Make sure ``pre-commit`` checks pass when committing (enforced in CI/CD build)
+5. Open a pull-request (PR) early for discussion
+6. Once the CI/CD build passes and PR is approved, **squash and rebase** your
+   commits
+7. Merge PR into ``master`` and **delete the branch**
+
+Things to Avoid
+^^^^^^^^^^^^^^^
+-  Merge in broken or commented out code
+-  Committing onto ``master`` directly
+-  Merge with conflicts (handle conflicts upon rebasing)
+
+Source: https://gist.github.com/jbenet/ee6c9ac48068889b0912
+
+Pre-commit
+~~~~~~~~~~
+The repository uses the ``pre-commit`` package to manage pre-commit hooks.
+These hooks help enforce QA standards and identify simple issues
+at the commit level before submitting code reviews.
+
+.. figure:: pre-commit-flow.svg
+   :alt: Pre-commit Flow Diagram
+
+   ``pre-commit`` Flow
+
+Helpful Commands
+^^^^^^^^^^^^^^^^
+
+Install into your conda environment
+   ::
+
+        conda activate e3sm_diags_env_dev
+        pre-commit install
+
+Automatically run all pre-commit hooks (just commit)
+    ::
+
+        git commit -m '...'
+
+    .. figure:: ../pre-commit-passing.png
+       :alt: pre-commit Output
+
+       ``pre-commit`` Output
+
+Manually run all pre-commit hooks
+   ::
+
+        pre-commit run --all-files.
+
+Run individual hook
+   ::
+
+        # Available hook ids: trailing-whitespace, end-of-file-fixer, check-yaml, black, isort, flake8, mypy
+        pre-commit run <hook_id>.
+
+Squash and Rebase Commits
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Before you merge a support branch back into ``master``, the branch is typically
+squashed down to a single buildable commit, and then rebased on top of the upstream ``master`` branch.
+
+-  Ensures build passes from the commit
+-  Cleans up Git history for easy navigation
+-  Makes collaboration and review process more efficient
+-  Makes handling conflicts from rebasing simple since you only have to
+   deal with conflicted commits
+
+Source:
+https://blog.carbonfive.com/always-squash-and-rebase-your-git-commits/
+
+Code Quality Assurance
+----------------------
+
+This project uses several tools for code formatting, linting, and type checking listed below.
+
+- Code Formatting: `black <https://black.readthedocs.io/en/stable/>`__
+- Linting: `flake8 <https://github.com/PyCQA/flake8#flake8>`__, `isort <https://pycqa.github.io/isort/>`__
+- Optional Type Checking: `mypy <http://mypy-lang.org/>`__
+
+You can run them as hooks manually/automatically when committing using ``pre-commit``, or manually through the terminal or IDE/text editor.
+
+Helpful Commands
+~~~~~~~~~~~~~~~~
+
+Run a tool
+    ::
+
+       # Available tool names: black, flake8, isort, mypy
+       <tool_name> .
+
+.. _ci-cd:
+
+Continuous Integration / Continuous Delivery (CI/CD)
+----------------------------------------------------
+
+This project uses `GitHub Actions <https://github.com/E3SM-Project/e3sm_diags/actions>`_ to run a CI/CD workflow on ``push`` and ``pull-request`` to ``master``.
```

Comment:
It runs on pushing directly to `master`? Do we want to be encouraging that?

Reply:
CI services have conditionals that will run the build on a `push` event to `master`, which is triggered from merging a PR. So in this context, we aren't pushing directly `master`. Also only the repo owner and added collaborators with write access can push to `master` (although we should probably add a branch protection rule to restrict it for everyone).

To make that description clearer, I will add "event" after ``push`` and ``pull_request``.